pkg/action/utils.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright SUSE LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package action
+
+import (
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+// SetNamespace sets the Namespace that should be used based on annotations or fallback to default
+// both on the action and on the storage.
+// if setDefault is true then it will just set the default namespace
+// This will read the chart annotations. If no annotations, it leave the existing ns in the action.
+// targetNS can be either the default namesapce (usually "default") or the namespace passed via
+// cli flag
+func SetNamespace(x interface{}, chart *chart.Chart, targetNS string, NamespaceFromFlag bool) {
```

Comment:
`s/NamespaceFromFlag/setDefault/` (from the function comment)?
I would prefer `setDefault` or something similar instead of `namespaceFromFlag`, as API users may not use any "ns from flag", nor the API has any, so it may confuse them.

Reply:
I used to have it as setDefault and changed it back to make it more clearer what it refer to... :rofl:

Reply:
reverted to `setDefault`